tests/src/main/scala/org/dhallj/tests/acceptance/AcceptanceSuccessSuite.scala

```diff
@@ -92,28 +96,35 @@ abstract class CachingExprOperationAcceptanceSuite(transformation: Expr => Expr)
   def compare(result: Expr, expected: Expr): Boolean = result.sameStructure(expected) && result.equivalent(expected)
 }
 
-class CachingTypeCheckingSuite(val base: String) extends CachingExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
-class TypeCheckingSuite(val base: String) extends ExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
-class AlphaNormalizationSuite(val base: String) extends CachingExprOperationAcceptanceSuite(_.alphaNormalize)
-class NormalizationSuite(val base: String) extends CachingExprOperationAcceptanceSuite(_.normalize)
-
-class HashingSuite(val base: String) extends ResolvingExprAcceptanceSuite[String] {
+class CachingTypeCheckingSuite(val base: String)
+    extends ExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
+    with CachedResolvingInput
+class TypeCheckingSuite(val base: String)
+    extends ExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
+    with ResolvingInput
+class AlphaNormalizationSuite(val base: String) extends ExprOperationAcceptanceSuite(_.alphaNormalize) with ParsingInput
+class NormalizationSuite(val base: String) extends ExprOperationAcceptanceSuite(_.normalize) with CachedResolvingInput
+class NormalizationUSuite(val base: String) extends ExprOperationAcceptanceSuite(_.normalize) with ParsingInput
```

Comment:
I'm not sure about the `U` suffix here. What does it mean?

Reply:
It means I wanted to call it `Unit` but that name was already taken 😂